packages/mui-material/src/InputBase/InputBase.js

```diff
@@ -77,7 +77,7 @@ const useUtilityClasses = (ownerState) => {
       fullWidth && 'fullWidth',
       focused && 'focused',
       formControl && 'formControl',
-      size === 'small' && 'sizeSmall',
+      size !== 'medium' && `size${capitalize(size)}`,
```

Comment:
Shouldn't it be:

```suggestion
      size && `size${capitalize(size)}`,
```
What if `size` is medium? Which class is applied then?

Reply:
fixed :)

did a rebase as well

Reply:
We don't add classes for default values, it just bloat the generated HTML without a clear value. The proposal is to add style overrides to the original class name and bump the specificity of a non default value is used. So I would say the initial implementation was a correct one.